src/pages/Facility/settings/locations/LocationList.tsx

```diff
@@ -1,49 +1,108 @@
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+import { Tabs, TabsList, TabsTrigger } from "@/components/ui/tabs";
 
-import Pagination from "@/components/Common/Pagination";
-import { CardGridSkeleton } from "@/components/Common/SkeletonLoading";
+import Page from "@/components/Common/Page";
+import { TableSkeleton } from "@/components/Common/SkeletonLoading";
+
+import useBreakpoints from "@/hooks/useBreakpoints";
 
 import query from "@/Utils/request/query";
+import { useView } from "@/Utils/useView";
 import { LocationList as LocationListType } from "@/types/location/location";
+import { getLocationFormLabel } from "@/types/location/location";
 import locationApi from "@/types/location/locationApi";
 
 import LocationSheet from "./LocationSheet";
-import { LocationCard } from "./components/LocationCard";
 
 interface Props {
   facilityId: string;
 }
 
+function buildLocationHierarchy(locations: LocationListType[]) {
+  const childrenMap = new Map<string, LocationListType[]>();
+  const topLevelLocations: LocationListType[] = [];
+
+  locations.forEach((location) => {
+    if (!location.parent || Object.keys(location.parent).length === 0) {
+      topLevelLocations.push(location);
+    } else {
+      const parentId = location.parent.id;
+      if (!childrenMap.has(parentId)) {
+        childrenMap.set(parentId, []);
+      }
+      childrenMap.get(parentId)?.push(location);
+    }
+  });
+
+  return { childrenMap, topLevelLocations };
+}
+
 export default function LocationList({ facilityId }: Props) {
   const { t } = useTranslation();
-  const [page, setPage] = useState(1);
   const [searchQuery, setSearchQuery] = useState("");
   const [selectedLocation, setSelectedLocation] =
     useState<LocationListType | null>(null);
   const [isSheetOpen, setIsSheetOpen] = useState(false);
-  const limit = 12;
+  const [activeTab, setActiveTab] = useView("locations", "list");
+  const [expandedRows, setExpandedRows] = useState<Record<string, boolean>>({});
+
+  const LIMIT = 1000;
 
   const { data, isLoading } = useQuery({
-    queryKey: ["locations", facilityId, page, limit, searchQuery],
-    queryFn: query.debounced(locationApi.list, {
+    queryKey: ["locations", facilityId, searchQuery],
+    queryFn: query(locationApi.list, {
       pathParams: { facility_id: facilityId },
       queryParams: {
-        parent: "",
-        offset: (page - 1) * limit,
-        limit,
+        limit: LIMIT,
         name: searchQuery || undefined,
       },
     }),
+    enabled: !!facilityId,
   });
 
+  const tableData = data?.results || [];
+  const { childrenMap, topLevelLocations } = useMemo(
+    () => buildLocationHierarchy(tableData),
+    [tableData],
+  );
+
+  const getChildren = useCallback(
+    (parentId: string) => {
+      const children = childrenMap.get(parentId) || [];
+      if (!searchQuery) return children;
+
+      return children.filter((loc) =>
+        loc.name.toLowerCase().includes(searchQuery.toLowerCase()),
```

Comment:
hmm 🤔  this wouldn't work since you are doing filtering for both top and children.

For example: for children to render, top level would have to match as well.